### PR TITLE
chore: add docs for resource servers

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to ensure RPs acting as a resource server have supporting documentation.

This commit:

* Adds appropriate clarifications to the docs for RPs acting as a resource server.

Closes FXA-7474

### CODE REVIEW

File: docs/relying-parties/tutorials/integrating-with-fxa.md [hunk 1]
```diff
@@ -74,31 +78,37 @@ Before starting integration, please send a request to fxa-staff[at]mozilla.com t
...
 
-## OAuth Integration
+## OpenID Connect Integration
+
+### Relying Party vs. Resource Server
+
+A Relying Party (RP) is an application or website that outsources its user authentication functionality to an OpenID Provider, defined as part of the OpenID specifications. The user will be prompted to login to the Relying Party with their Firefox Account, and is able to see the login as a distinct [Connected Service][connected-services] in FxA Settings. OpenID Connect contains many specifications for a variety of identity exchange and authorization, based on the OAuth 2.0 framework of specifications.
+
+In contrast, a Resource Server is an OAuth 2.0 term for an API server. A Relying Party may also act as a Resource Server, with a user never seeing a direct Relying Party login. Users that access a resource server with OAuth tokens will not see their access as a distinct Relying Party login on the FxA Settings page. Services that are accessed via Firefox (Sync, Relay, etc.) using an existing OAuth token are all examples of Resource Servers. Their access shows up in FxA Settings as the browser session the user is logged into.
```
Comment (chenba) [inline docs/relying-parties/tutorials/integrating-with-fxa.md:87]: Relay is actually listed as Relay on the connected services.  Sync is only special case.

> using an existing OAuth token

I think the word "existing" here can be confusing.  It reads as if it's referring to some particular existing token.

File: docs/relying-parties/tutorials/integrating-with-fxa.md [hunk 2]
```diff
@@ -74,31 +78,37 @@ Before starting integration, please send a request to fxa-staff[at]mozilla.com t
...
 :::note
 You are encouraged to use [our staging servers](https://accounts.stage.mozaws.net/) to develop against.  Our staging server has a persistent database and changes made there are saved.  Spot testing and some new accounts are fine but you are expected to clean up any significant amount of data (eg. accounts made from automated testing).  If you're using PyFxA [here is an example](https://pypi.org/project/PyFxA/#testing-email-addresses) using `destroy_account()`.
 :::
 
-0. Review the [OAuth 2.0][oauth] documentation.
+0. Review the [OpenID Connect][how connect works] and [OAuth 2.0][oauth] documentation.
 0. Register for staging OAuth credentials by filing a [deployment bug][deployment-bug]. See [OAuth credentials](#oauth-credentials).
 0. Your development servers should point to: `https://oauth.stage.mozaws.net`.
-0. User authentication follows the [OAuth 2.0][oauth] protocol.
+0. User authentication follows the [OpenID Connect][openidconnect] protocol.
```
Comment (chenba) [inline docs/relying-parties/tutorials/integrating-with-fxa.md:98]: Treating FxA as an OAuth provider should still work.  Or do we simply want to discourage that?

File: docs/relying-parties/tutorials/integrating-with-fxa.md (hunk 1 above)
Comment (bbangert) [reply docs/relying-parties/tutorials/integrating-with-fxa.md:87]: Relay is not always listed anymore as they now operate as a Resource Server. It is now possible to use Relay via Firefox without it appearing in Connected Services.

File: docs/relying-parties/tutorials/integrating-with-fxa.md (hunk 2 above)
Comment (bbangert) [reply docs/relying-parties/tutorials/integrating-with-fxa.md:98]: OpenID Connect is built on OAuth, so using the higher level term still encompasses OAuth.

File: docs/relying-parties/tutorials/integrating-with-fxa.md [hunk 3]
```diff
@@ -4,22 +4,26 @@ title: Integration with FxA
...
+Firefox Accounts integration is available for Mozilla groups on request. This integration is handled using [OpenID Connect (OIDC)][openidconnect], [OAuth 2.0][oauth], and [webhooks][webhook] for authentication, authorization, and receiving events regarding FxA users. Integrations with FxA assume the role of a [Relying Party (RP)][relying-party] and/or a [Resource Server (RS)][resource server] depending on the type of integration, with FxA assuming the role of an OpenID Provider.
 
 :::note
-This tutorial will help you integrate with Firefox accounts but there are [additional requirements a relying party is expected to fulfill and maintain](/relying-parties/reference/integration-requirements).  Please ensure you're in compliance with expectations to continue uninterrupted service.
+This tutorial will help you integrate with Firefox Accounts but there are [additional requirements a relying party is expected to fulfill and maintain](/relying-parties/reference/integration-requirements).  Please ensure you're in compliance with expectations to continue uninterrupted service.
```
Comment (clouserw) [inline docs/relying-parties/tutorials/integrating-with-fxa.md:14]: Not a big deal, but this shouldn't be capitalized

File: docs/relying-parties/tutorials/integrating-with-fxa.md [hunk 4]
```diff
@@ -109,6 +119,8 @@ You are encouraged to use [our staging servers](https://accounts.stage.mozaws.ne
 
 ### OAuth Credentials
 
+OAuth Client Credentials are needed for each application accessing FxA. For example, a website that users can login to with mobile applications for iOS and Android should request 3 sets of OAuth credentials, one for each mobile app/platform, and one for the web service.
```
Comment (clouserw) [inline docs/relying-parties/tutorials/integrating-with-fxa.md:122]: nice

File: docs/relying-parties/tutorials/integrating-with-fxa.md (hunk 1 above)
Comment (clouserw) [reply docs/relying-parties/tutorials/integrating-with-fxa.md:87]: I think reorganizing the first 3 sentences could help clarify.  How does this sound?
> A Relying Party may also act as a Resource Server, an OAuth 2.0 term for an API server.  In that case a user will never see a direct Relying Party login and will not see a distinct row in their Connected Devices list on the FxA Settings page. [Services that are....]

**Review by clouserw — APPROVED** [review]

File: docs/relying-parties/tutorials/integrating-with-fxa.md (hunk 1 above)
Comment (bbangert) [reply docs/relying-parties/tutorials/integrating-with-fxa.md:87]: @chenba Agreed, rephrasing, how does this look?

**Review by chenba — APPROVED** [review]

**Pull request merged**